components/ciphertool/src/main/java/org/wso2/ciphertool/CipherTool.java

```diff
@@ -58,32 +75,58 @@ public class CipherTool {
 
         initialize(args);
-        Cipher cipher = KeyStoreUtil.initializeCipher();
-        if (System.getProperty(Constants.CONFIGURE) != null &&
-            System.getProperty(Constants.CONFIGURE).equals(Constants.TRUE)) {
+        KeyStore keyStore = KeyStoreUtil.getKeyStore();
+        CipherMode cipherMode = CipherFactory.createCipher(keyStore);
+        if (Constants.TRUE.equals(System.getProperty(Constants.CONFIGURE))) {
             File deploymentTomlFile = new File(Utils.getDeploymentFilePath());
             if (deploymentTomlFile.exists()) {
                 Map<String, String> secretMap = Utils.getSecreteFromConfiguration(Utils.getDeploymentFilePath());
                 for (Map.Entry<String, String> entry : secretMap.entrySet()) {
                     String key = entry.getKey();
                     String value = Utils.getUnEncryptedValue(entry.getValue());
                     if (StringUtils.isNotEmpty(value)) {
-                        String encryptedValue = Utils.doEncryption(cipher, value);
+                        String encryptedValue = cipherMode.doEncryption(value);
                         secretMap.replace(key, encryptedValue);
                     }
                 }
                 updateDeploymentConfigurationWithEncryptedKeys(secretMap);
             } else {
                 loadXpathValuesAndPasswordDetails();
                 secureVaultConfigTokens();
-                encryptCipherTextFile(cipher);
+                encryptCipherTextFile(cipherMode);
             }
             Utils.writeToSecureConfPropertyFile();
-        } else if (System.getProperty(Constants.CHANGE) != null &&
-                   System.getProperty(Constants.CHANGE).equals(Constants.TRUE)) {
-            changePassword(cipher);
+        } else if (Constants.TRUE.equals(System.getProperty(Constants.ROTATE))) {
+            String oldAlias  = System.getProperty(Constants.OLD_KEY_ALIAS);
+            if (StringUtils.isBlank(oldAlias)) {
+                throw new CipherToolException(Constants.OLD_KEY_ALIAS + " parameter is required for key rotate mode.");
+            }
+            CipherMode oldCipherMode = isSymmetricKey(oldAlias, keyStore)
+                    ? new SymmetricCipher(keyStore, oldAlias) : new AsymmetricCipher(keyStore, oldAlias);
+            File deploymentTomlFile = new File(Utils.getDeploymentFilePath());
+            if (deploymentTomlFile.exists()) {
+                Map<String, String> secretMap = Utils.getSecreteFromConfiguration(Utils.getDeploymentFilePath());
+                for (Map.Entry<String, String> entry : secretMap.entrySet()) {
+                    String key = entry.getKey();
+                    String oldEncryptedValue = Utils.getEncryptedValue(entry.getValue());
+                    if (StringUtils.isNotEmpty(oldEncryptedValue)) {
+                        String value = oldCipherMode.doDecryption(oldEncryptedValue);
+                        secretMap.replace(key, value);

```

Comment:
Why do you need map decrypted and encrypted values to the same key if the value is not empty ? There is no meaning to the key value pair if the value is empty. am I correct ?